taichi/codegen/cuda/codegen_cuda.cpp

```diff
@@ -105,11 +105,16 @@ class TaskCodeGenCUDA : public TaskCodeGenLLVM {
 
     std::string formats;
     size_t num_contents = 0;
-    for (auto const &content : stmt->contents) {
+    TI_ASSERT(stmt->contents.size() == stmt->formats.size());
```

Comment:
If there's no CHI IR Pass that transforms `contents` and `formats`, we can migrate this validation to `irpass::type_check()` or even `FrontendPrintStmt::type_check()` - these are common places for verifications.

Reply:
Type checking was migrated to FrontendPrintStmt::type_check()